lib/ansible/modules/network/f5/bigip_command.py

```diff
@@ -0,0 +1,379 @@
+
+        result.update(**self.changes.to_return())
+        result.update(dict(changed=False))
+        return result
+
+    def execute(self):
+        warnings = list()
+
+        commands = self.parse_commands(warnings)
+
+        wait_for = self.want.wait_for or list()
+        retries = self.want.retries
+
+        conditionals = [Conditional(c) for c in wait_for]
+
+        while retries > 0:
+            responses = self.execute_on_device(commands)
+
+            for item in list(conditionals):
+                if item(responses):
+                    if self.want.match == 'any':
+                        return item
+                    conditionals.remove(item)
+
+            if not conditionals:
+                break
+
+            time.sleep(self.want.interval)
+            retries -= 1
+        else:
+            failed_conditions = [item.raw for item in conditionals]
+            errmsg = 'One or more conditional statements have not been satisfied'
+            raise FailedConditionsError(errmsg, failed_conditions)
+
+        self.changes = Parameters({
+            'stdout': responses,
+            'stdout_lines': self._to_lines(responses),
+            'warnings': warnings
+        })
+
+    def parse_commands(self, warnings):
+        results = []
+        commands = list(deque(set(self.want.commands)))
+        spec = dict(
+            command=dict(key=True),
+            output=dict(
+                default='text',
+                choices=['text', 'one-line']
+            ),
+        )
+
+        transform = ComplexList(spec, self.client.module)
+        commands = transform(commands)
+
+        for index, item in enumerate(commands):
+            if not self._is_valid_mode(item['command']):
+                warnings.append(
+                    'Using "write" commands is not idempotent. You should use '
+                    'a module that is specifically made for that. If such a '
+                    'module does not exist, then please file a bug. The command '
+                    'in question is "%s..."' % item['command'][0:40]
+                )
+            if item['output'] == 'one-line' and 'one-line' not in item['command']:
+                item['command'] += ' one-line'
+            elif item['output'] == 'text' and 'one-line' in item['command']:
+                item['command'] = item['command'].replace('one-line', '')
+            results.append(item)
+        return results
+
+    def execute_on_device(self, commands):
+        responses = []
+        for item in to_list(commands):
+            output = self.client.api.tm.util.bash.exec_cmd(
+                'run',
+                utilCmdArgs='-c "{0}"'.format(item['command'])
+            )
+            if hasattr(output, 'commandResult'):
+                responses.append(str(output.commandResult))
+        return responses
+
+
+class ArgumentSpec(object):
+    def __init__(self):
+        self.supports_check_mode = True
```

Comment:
check_mode is True but I don't see any check done around this. Please correct me if I am wrong here.